src/controllers/authController.js

```diff
@@ -5,6 +5,36 @@ const CustomError = require('../errors');
 const { attachCookiesToResponse, createTokenUser } = require('../utils');
 
 const register = async (req, res) => {
+/* 
+    #swagger.summary = 'Register a new user'
+    #swagger.parameters['user'] = {
+      in: 'body',
+      description: 'User registration information',
+      required: true,
+      schema: {
+        type: 'object',
+        properties: {
+          name: { type: 'string' },
+          email: { type: 'string' },
+          password: { type: 'string' },
+          username: { type: 'string' }
+        }
+      }
```

Comment:
The swagger-autogen library wants us to write the schema not in the formal OpenApi/Swagger spec way, but rather exactly as the example object you would want to see in the docs, and then it will figure out the tyeps from there: [https://github.com/swagger-autogen/swagger-autogen#schema-and-definitions](https://github.com/swagger-autogen/swagger-autogen#schema-and-definitions)

```suggestion
      schema: {
        $name: "John Doe",
        $email: "jdoe@example.com",
        $password: "supersecretpassword",
        $username: "jdoe99"
      }
```

We use the dollar sign in front of the field to denote that it is required